src/pykeen/triples/triples_factory.py

```diff
@@ -610,38 +660,36 @@ def new_with_restriction(
             A new triples factory, which has only a subset of the triples containing the entities and relations of
             interest. The label-to-ID mapping is *not* modified.
         """
-        keep_mask: Optional[torch.BoolTensor] = None
-
+        # prepare metadata
         extra_metadata = {}
-        # Filter for entities
         if entities is not None:
-            if any(isinstance(e, str) for e in entities):
-                raise ValueError(f"{self.__class__} does not support label-based restriction.")
-            entities = cast(Collection[int], entities)
             extra_metadata["entity_restriction"] = entities
-            keep_mask = self.get_mask_for_entities(entities=entities, invert=invert_entity_selection)
-            remaining_entities = self.num_entities - len(entities) if invert_entity_selection else len(entities)
+            entities = self.entities_to_ids(entities=entities)
+            remaining_entities = (self.num_entities - len(entities)) if invert_entity_selection else len(entities)
             logger.info(f"keeping {format_relative_comparison(remaining_entities, self.num_entities)} entities.")
-
-        # Filter for relations
         if relations is not None:
-            if any(isinstance(r, str) for r in relations):
-                raise ValueError(f"{self.__class__} does not support label-based restriction.")
-            relations = cast(Collection[int], relations)
             extra_metadata["relation_restriction"] = relations
-            relation_mask = self.get_mask_for_relations(relations=relations, invert=invert_relation_selection)
-            remaining_relations = self.num_relations - len(relations) if invert_entity_selection else len(relations)
+            relations = self.relations_to_ids(relations=relations)
+            remaining_relations = (self.num_relations - len(relations)) if invert_entity_selection else len(relations)
```

Comment:
should this be `invert_relation_selection`?

Reply:
yes, good catch!

Reply:
yes, good catch!